server/tests/test_notebook_view.py

```diff
@@ -151,6 +151,72 @@ def test_tryit_view(client, fake_user, logged_in, iomd):
         assert len(response.redirect_chain) == 0
 
 
+def test_from_template_view_reject_get_request(client):
+
+    resp = client.get(reverse("from-template"))
+    assert resp.status_code == 405
+
+
+def test_from_template_view_reject_no_iomd(client):
+
+    resp = client.post(reverse("from-template"), {"iomd": "",},)
+    assert resp.status_code == 400
+    assert resp.content == b"Must specify iomd template"
+
+
+def test_from_template_view_file_too_big(client, settings):
+    settings.MAX_FILE_SIZE = 8
+    with tempfile.NamedTemporaryFile(mode="w+") as f1, tempfile.NamedTemporaryFile(mode="w+") as f2:
```

Comment:
It would probably be better to use a text i/o object here, instead of a temporary file:

https://docs.python.org/3/library/io.html#text-i-o

Reply:
> It would probably be better to use a text i/o object here, instead of a temporary file:
> 
> https://docs.python.org/3/library/io.html#text-i-o
> 
> Not strictly necessary to change (it's not that big a deal), just letting you know.

great way to solve this problem. since we don't really need file names. thank you